### DISCUSSION TITLE
URI, URL, URN에 대한 설명

### DISCUSSION THREAD

**Discussion by GIVEN53** [2024-woowacourse-study/level-interview#191] [Network]
.

**Comment by alstn113** [comment]
### URI (Uniform Resource Identifier)
URI는 인터넷에서 자원을 식별하기 위한 문자열입니다. URI는 URL과 URN을 포함하는 상위 개념입니다. 
즉, 특정 자원을 식별하기 위한 포괄적인 방법을 제공하며, 자원의 위치나 이름을 나타낼 수 있습니다.

### URL (Uniform Resource Locator)
URL은 URI의 한 형태로, 인터넷 상에서 자원의 위치를 나타내는 방식입니다. 
자원이 어디에 있는지를 설명하는데 사용되며, 자원에 접근하기 위한 프로토콜을 포함합니다. 
예를 들어, 웹페이지의 URL은 해당 페이지가 위치한 서버의 주소와 접근 방법(예: HTTP)을 포함합니다.
- 예시: `https://www.example.com/path/to/resource`

### URN (Uniform Resource Name)
URN은 URI의 또 다른 형태로, 자원의 위치와 상관없이 자원의 이름을 식별하는 방식입니다.
자원의 위치가 변하더라도 동일한 식별자를 유지할 수 있게 합니다. 
특정 스키마를 따르며, 자원에 대한 영구적인 식별자를 제공합니다.
- 예시: `urn:isbn:0451450523` (특정 책의 ISBN 번호)

<img width="624" alt="image" src="https://github.com/2024-woowacourse-study/level-interview/assets/75781414/fb71b6e2-cc61-4518-9370-1285271e3a46">

참고: https://inpa.tistory.com/entry/WEB-🌐-URL-URI-차이

**Comment by 3Juhwan** [comment]
**URI(Uniform Resource Identifier)**
- 통합 자원 식별자로, 인터넷에 있는 자원을 나타내는 유일한 주소
- URI의 하위개념으로 URL, URN이 있다

**URL(Uniform Resource Locator)**
- 네트워크 상에서 자원의 위치를 고유하게 식별할 수 있는 주소
- 흔히 웹 사이트 주소로 알고 있지만, URL은 웹 사이트 주소뿐만 아니라 컴퓨터 네트워크상의 자원 위치를 파악하여 데이터를 찾아낼 수 있습니다. 

**URN (Uniform Resource Name)**
- 자원을 식별하는 데 사용되는 이름

현재는 용어를 구분하지 않고 혼용해서 사용합니다. URI는 URL과 URN을 포함하는 상위 개념, URL은 자원의 위치를, URN은 자원의 이름을 나타낸다 정도만 기억하면 될 것 같습니다. 

<img src="https://github.com/2024-woowacourse-study/level-interview/assets/64410384/c81c74a6-e878-44e2-bce7-bdb8300faf2b" width="400">

**Reply by le2sky** [reply]
![image](https://github.com/user-attachments/assets/6fabcb82-3a28-4de6-9870-3c466a61b775)
